resources/shaders/xray_composite.shader

```diff
@@ -50,10 +51,12 @@ fragment =
 
         result = layer0 * layer0.a + result * (1.0 - layer0.a);
 
-        float intersection_count = (texture2D(u_layer2, v_uvs).r * 255.0) / 5.0;
-        if(mod(intersection_count, 2.0) == 1.0)
+        float intersection_count = texture2D(u_layer2, v_uvs).r * 51; // (1 / .02) + 1 (+1 magically fixes issues with high intersection count models)
+        float rest = mod(intersection_count + .01, 2.0);
+        if (rest > 1.0 && rest < 1.5 && intersection_count < 49)
         {
-            result = u_error_color;
+            vec2 scaling = textureSize(u_layer0, 0) / textureSize(u_xray_error, 0);
```

Comment:
Note that `textureSize` was added to GLSL 1.30, which is part of OpenGL 3.0 ... thus it'll probably not work on OpenGL 2.1 systems (which is what the not-core-4.1 shaders use)!

Reply:
That's a lot of different version numbers!

What exactly needs to be done?

Reply:
An alternative to using textureSize() needs to be found, such as eg passing the texturesize as a uniform.

Reply:
Ok. I don't have time for that atm.

Reply:
It would probably be more performant anyway to pass this scaling factor as a uniform once per frame instead of recalculating it for every pixel.

Reply:
@BagelOrb, I have made a fix on top of this branch, here: https://github.com/Ultimaker/Cura/pull/7363

Reply:
@smartavionics could you test this branch (and my PR) with your Raspberry Pi build?

Reply:
That commit got merged into here, so no need to test your branch separately from this PR.

Let's hope the Cura team can test this.